### ISSUE TITLE
fieldset check should not prioritise fieldset over aria group

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#1315]
This shouldn't fail the `checkboxgroup` rule:

```html
<fieldset>
	<div role="group" aria-label="Group label">
		<input type="checkbox" name="foo" />
		<input type="checkbox" name="foo" />
	</div>
</fieldset>
```

Might be best to simply put the ARIA grouping code into its own check, rather than have it be part of the fieldset check. We need to think through the implications of that, but it would sure clear up the code in that check.

**Comment by dylanb** [comment]
doesn't seem high priority to me. Also needs some thought, what about:

```
<fieldset>
        <legend>I don't like groups</legend>
	<div role="group" aria-label="Group label">
		<input type="checkbox" name="foo" />
		<input type="checkbox" name="foo" />
	</div>
</fieldset>
```
